soccer/src/soccer/radio/network_radio.cpp

```diff
@@ -26,54 +28,44 @@ void NetworkRadio::start_receive() {
-        const auto& intent = intents[shell];
-        if (!intent.is_active) {
-            continue;
-        }
-
-        const auto& setpoint = setpoints[shell];
-
-        // Build the control packet for this robot.
-        std::array<uint8_t, rtp::HeaderSize + sizeof(rtp::RobotTxMessage)>& forward_packet_buffer =
-            send_buffers_[shell];
-
-        auto* header = reinterpret_cast<rtp::Header*>(&forward_packet_buffer[0]);
-        fill_header(header);
-
-        auto* body =
-            reinterpret_cast<rtp::RobotTxMessage*>(&forward_packet_buffer[rtp::HeaderSize]);
-
-        ConvertTx::to_rtp(intent, setpoint, shell, body);
-
-        // Fetch the connection
-        auto maybe_connection = connections_.at(shell);
-
-        // If there exists a connection, we can send.
-        if (maybe_connection) {
-            const RobotConnection& connection = maybe_connection.value();
-            // Check if we've timed out.
-            if (RJ::now() + kTimeout < connection.last_received) {
-                // Remove the endpoint from the IP map and the connection list
-                assert(robot_ip_map_.erase(connection.endpoint) == 1);
-                connections_.at(shell) = std::nullopt;
-            } else {
-                // Send to the given IP address
-                const udp::endpoint& robot_endpoint = connection.endpoint;
-                socket_.async_send_to(
-                    boost::asio::buffer(forward_packet_buffer), robot_endpoint,
-                    [](const boost::system::error_code& error, std::size_t num_bytes) {
-                        // Handle errors.
-                        if (static_cast<bool>(error)) {
-                            SPDLOG_ERROR(  // NOLINT(bugprone-lambda-function-name)
-                                "Error sending: {}.", error);
-                        }
-                    });
-            }
+void NetworkRadio::send(int robot_id, const rj_msgs::msg::MotionSetpoint& motion,
+                        const rj_msgs::msg::ManipulatorSetpoint& manipulator) {
+    // Build the control packet for this robot.
+    std::array<uint8_t, rtp::HeaderSize + sizeof(rtp::RobotTxMessage)>& forward_packet_buffer =
+        send_buffers_[robot_id];
+
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+    auto* header = reinterpret_cast<rtp::Header*>(&forward_packet_buffer[0]);
+    fill_header(header);
+
+    // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+    auto* body = reinterpret_cast<rtp::RobotTxMessage*>(&forward_packet_buffer[rtp::HeaderSize]);
+
+    ConvertTx::ros_to_rtp(manipulator, motion, robot_id, body);
+
+    // Fetch the connection
+    auto maybe_connection = connections_.at(robot_id);
+
+    // If there exists a connection, we can send.
+    if (maybe_connection) {
+        const RobotConnection& connection = maybe_connection.value();
+        // Check if we've timed out.
+        if (RJ::now() + kTimeout < connection.last_received) {
```

Comment:
I don't quite get this line. now + a time will always be in the future, and the last received time will always be in the past? So will this ever be true?